client-server-client/src/main/scala/zio/raft/client/RaftClient.scala

```diff
@@ -0,0 +1,536 @@
+   */
+  def acknowledgeServerRequest(requestId: RequestId): UIO[Unit] = 
+    for {
+      ack = ServerRequestAck(requestId)
+      _ <- zmqTransport.sendMessage(ack).orDie
+      _ <- ZIO.logDebug(s"Acknowledged server request: $requestId")
+    } yield ()
+}
+
+object RaftClient {
+  
+  /**
+   * Create and start a RaftClient.
+   * The client automatically connects to the cluster on creation.
+   */
+  def make(addresses: List[String], capabilities: Map[String, String]): ZIO[Scope, Throwable, RaftClient] = 
+    for {
+      // Create and validate configuration
+      config = ClientConfig.make(addresses, capabilities)
+      validatedConfig <- ClientConfig.validated(config).mapError(new IllegalArgumentException(_))
+      
+      // Create ZMQ transport
+      zmqTransport <- createZmqTransport(validatedConfig)
+      
+      // Create action queue for stream-based processing
+      actionQueue <- Queue.unbounded[ClientAction]
+      
+      // Create client
+      client = new RaftClient(
+        zmqTransport,
+        validatedConfig,
+        actionQueue
+      )
+      
+      // Start the main processing loop (forked)
+      _ <- startMainLoop(client, zmqTransport, validatedConfig, actionQueue).fork
+      
+      // Automatically initiate connection
+      _ <- initiateConnection(zmqTransport, validatedConfig, actionQueue)
+      
+    } yield client
+  
+  /**
+   * Initiate connection - just sends CreateSession message.
+   * Does not wait for response, which will be handled by the stream.
+   */
+  private def initiateConnection(
+    transport: ZmqClientTransport,
+    config: ClientConfig,
+    actionQueue: Queue[ClientAction]
+  ): Task[Unit] = 
+    for {
+      nonce <- Nonce.generate()
+      createMessage = CreateSession(config.capabilities, nonce)
+      _ <- transport.sendMessage(createMessage)
+      _ <- ZIO.logInfo("CreateSession sent, transitioning to Connecting")
+    } yield ()
+  
+  private def createZmqTransport(config: ClientConfig): UIO[ZmqClientTransport] = 
+    ZIO.succeed(new ZmqClientTransportStub())
+  
+  /**
+   * Main processing loop using unified stream with fold.
+   * Each fold iteration returns a new ClientState.
+   */
+  private def startMainLoop(
+    client: RaftClient,
+    transport: ZmqClientTransport,
+    config: ClientConfig,
+    actionQueue: Queue[ClientAction]
+  ): Task[Unit] = {
+    
+    // Create unified stream merging all event sources
+    val actionStream = ZStream.fromQueue(actionQueue)
+      .map(StreamEvent.Action(_))
+    
+    val messageStream = transport.incomingMessages
+      .map(StreamEvent.ServerMsg(_))
+    
+    val keepAliveStream = ZStream.tick(config.keepAliveInterval)
+      .map(_ => StreamEvent.KeepAliveTick)
+    
+    val unifiedStream = actionStream
+      .merge(messageStream)
+      .merge(keepAliveStream)
+    
+    // Initial state: Connecting with first address
+    val initialState = ClientState.Connecting(
```

Comment:
this should not be here, should be accepted as param. For example, the nonce is wrong.